### PR TITLE
Fall back to Python2 when detect SyntaxError on calling pydep

### LINKED ISSUE #54: Reddit Python app not detected
https://sourcegraph.com/github.com/reddit/reddit/-/blob/r2/setup.py defines the pip package that comprises the Reddit web application. It is not detected by the scanner so no Python files are linkedin Reddit.

### PR DESCRIPTION
**Pull request by unknwon**
Should fix #54 #52 #41 , depends on https://github.com/sourcegraph/pydep/pull/19

### CODE REVIEW

File: grapher/scan.py [hunk 1]
```diff
@@ -66,10 +65,11 @@ def source_files_for_pip_unit(unit_dir: str) -> Tuple[List[str], List[str]]:
         modules.extend(metadata['py_modules'])
 
     # Indicate whether this unit is in root direcotry of repository.
+    unit_dir = metadata['rootdir']
     is_root_dir = unit_dir == "."
     included_tests = False
     files = []
-    for module in modules:
+    for module in modules:
```
Comment (beyang) [inline grapher/scan.py]: remove unnecessary whitespace

File: grapher/pydepwrap.py [hunk 2]
```diff
@@ -17,12 +17,30 @@
 def setup_dirs(rootdir: str) -> List[str]:
     return pydep.setup_py.setup_dirs(rootdir)
 
-
-def setup_info_dir(setup_dir: str) -> Any:
-    info, err = pydep.setup_py.setup_info_dir(setup_dir)
+# pydep_in_python2 uses Python2 to run pydep and returns parsed JSON object.
+def pydep_in_python2(cmd: str, dir: str) -> Any:
+    rootdir = os.path.dirname(os.path.dirname(__file__))
+    if os.environ.get('SRCLIBPATH') is not None: # Not None only in Docker container
```
Comment (beyang) [inline grapher/pydepwrap.py]: `SRCLIBPATH` might be set outside the context of a Docker container, too. Is this necessary for running inside a Docker container? If so, could you explain why in the comments?

File: grapher/pydepwrap.py [hunk 3]
```diff
@@ -17,12 +17,30 @@
 def setup_dirs(rootdir: str) -> List[str]:
     return pydep.setup_py.setup_dirs(rootdir)
 
-
-def setup_info_dir(setup_dir: str) -> Any:
-    info, err = pydep.setup_py.setup_info_dir(setup_dir)
+# pydep_in_python2 uses Python2 to run pydep and returns parsed JSON object.
```
Comment (beyang) [inline grapher/pydepwrap.py]: Add a comment stating that this is a bit of a hack for posterity's sake. E.g., "This is a KLUDGE/HACK because it introduces an implicit dependency on srclib."

**Comment by beyang** [comment]
Minor comments, otherwise LGTM, thanks!

File: grapher/pydepwrap.py (hunk 2 above)
Comment (Unknwon) [inline grapher/pydepwrap.py]: > Is this necessary for running inside a Docker container? 

No... just Docker container for srclib-python has this env var set and we could use it. I'll remove the comment here.

**Comment by Unknwon** [comment]
New commit pushed.

**Comment by beyang** [comment]
@Unknwon I tried running this locally and wasn't able to get it to build due to it detecting the wrong rootdir. I tried a different approach here: https://github.com/sourcegraph/srclib-python/commits/python2-pydep-beyang. This makes the scan phase run correctly, but it still misses the modules declared in the setup.py. I have a test case here: https://github.com/sgtest/python27-syntax.

**Comment by Unknwon** [comment]
>I tried running this locally

Which `local` are you referring to? Local machine or Docker?

**Comment by beyang** [comment]
```
beyang@huck:~/src/github.com/sgtest/python27-syntax$ srclib config
Traceback (most recent call last):
  File "/home/beyang/.srclib/github.com/sourcegraph/srclib-python/.env/local/lib/python3.5/site-packages/grapher/pydepwrap.py", line 37, in setup_info_dir
    info, err = pydep.setup_py.setup_info_dir(setup_dir)
  File "/home/beyang/.srclib/github.com/sourcegraph/srclib-python/.env/local/lib/python3.5/site-packages/pydep/setup_py.py", line 41, in setup_info_dir
    return setup_info(setupfile), None
  File "/home/beyang/.srclib/github.com/sourcegraph/srclib-python/.env/local/lib/python3.5/site-packages/pydep/setup_py.py", line 76, in setup_info
    runpy.run_path(path.basename(setupfile), run_name='__main__')
  File "/usr/lib/python3.5/runpy.py", line 238, in run_path
    code, fname = _get_code_from_file(run_name, path_name)
  File "/usr/lib/python3.5/runpy.py", line 213, in _get_code_from_file
    code = compile(f.read(), fname, 'exec')
  File "setup.py", line 3
    print "python-2-style print statement in setup.py"
                                                     ^
SyntaxError: Missing parentheses in call to 'print'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.5/runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/beyang/.srclib/github.com/sourcegraph/srclib-python/.env/local/lib/python3.5/site-packages/srclib-python.py", line 34, in <module>
    main()
  File "/home/beyang/.srclib/github.com/sourcegraph/srclib-python/.env/local/lib/python3.5/site-packages/srclib-python.py", line 23, in main
    scan(os.getcwd())
  File "/home/beyang/.srclib/github.com/sourcegraph/srclib-python/.env/local/lib/python3.5/site-packages/grapher/scan.py", line 160, in scan
    for pkg in find_pip_pkgs(diry):
  File "/home/beyang/.srclib/github.com/sourcegraph/srclib-python/.env/local/lib/python3.5/site-packages/grapher/scan.py", line 50, in find_pip_pkgs
    setup_dict = pydepwrap.setup_info_dir(setup_dir)
  File "/home/beyang/.srclib/github.com/sourcegraph/srclib-python/.env/local/lib/python3.5/site-packages/grapher/pydepwrap.py", line 42, in setup_info_dir
    return pydep_in_python2("info", setup_dir)
  File "/home/beyang/.srclib/github.com/sourcegraph/srclib-python/.env/local/lib/python3.5/site-packages/grapher/pydepwrap.py", line 28, in pydep_in_python2
    process = subprocess.Popen([python, pydep, cmd, dir], stdout=subprocess.PIPE)
  File "/usr/lib/python3.5/subprocess.py", line 950, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.5/subprocess.py", line 1544, in _execute_child
    raise child_exception_type(errno_num, err_msg)
FileNotFoundError: [Errno 2] No such file or directory: '/home/beyang/.srclib/sourcegraph.com/sourcegraph/srclib-python/.env/bin/python2.7'
Traceback (most recent call last):
  File "/home/beyang/.srclib/github.com/sourcegraph/srclib-python/.bin/srclib-python", line 13, in <module>
    subprocess.run(args, check=True)
  File "/usr/lib/python3.5/subprocess.py", line 711, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['/home/beyang/.srclib/github.com/sourcegraph/srclib-python/.env/bin/python3.5', '-m', 'srclib-python', 'scan']' returned non-zero exit status 1
failed to scan for source units: scanner [/home/beyang/.srclib/github.com/sourcegraph/srclib-python/.bin/srclib-python scan]: parsing the STDOUT of the scanner failed with: EOF
FAILED: srclib config (failed to scan for source units: scanner [/home/beyang/.srclib/github.com/sourcegraph/srclib-python/.bin/srclib-python scan]: parsing the STDOUT of the scanner failed with: EOF)
```

**Comment by beyang** [comment]
This is on the test repository I mentioned in my previous comment. Can you try that locally to see if you can reproduce?

**Comment by beyang** [comment]
@Unknwon cleaned up and merged into trunk as https://github.com/sourcegraph/srclib-python/commit/f83063c1521b183c4d02a1fadc38cfe114d1482a
